cmd/snap-confine/ns-support.c

```diff
@@ -287,20 +287,38 @@ static bool should_discard_current_ns(dev_t base_snap_dev)
 	if (mi == NULL) {
 		die("cannot parse mountinfo of the current process");
 	}
+	/* We are looking for a mount entry matching the device ID of the base
+	 * snap. We need to take these cases into account:
+	 * 1) In the typical case, this will be mounted on the "/" directory.
+	 * 2) If the root directory is a tmpfs, the base snap would be mounted
+	 *    under /usr.
+	 * 3) If the snap has a layout that adds directories or files directly
+	 *    under /usr, a writable mimic will be created: /usr will be a tmpfs,
+	 *    with all of the original directory entries inside of /usr being
+	 *    bind-mounted onto mount-points created into the tmpfs.
+	 * In light of the above, we do ignore all tmpfs entries and accept that
+	 * our base snap might be mounted under /, /usr, or anywhere under /usr.
+	 */
 	for (mie = sc_first_mountinfo_entry(mi); mie != NULL;
 	     mie = sc_next_mountinfo_entry(mie)) {
-		if (!sc_streq(mie->mount_dir, "/")) {
+		// If the "/" directory is a tmpfs, then we'll look at the /usr
+		// directory
```

Comment:
the comment is probably no longer needed
```suggestion
```

Reply:
Oops, removed!